src/higlass/tilesets.py

```diff
@@ -74,10 +75,14 @@ def remote(uid: str, server: str = "https://higlass.io/api/v1", **kwargs):
 def hash_absolute_filepath_as_default_uid(
     fn: typing.Callable[[str, str], LocalTileset]
 ):
-    def wrapper(filepath: str, uid: None | str = None):
+    def wrapper(filepath: Union[str, IO[bytes]], uid: None | str = None):
```

Comment:
To be more explicit, maybe:

```suggestion
    def wrapper(filepath_or_buffer: Union[str, IO[bytes]], uid: None | str = None):
```

Reply:
Also, `Union` is deprecated in favor of `str | IO[Bytes]`. You will need `from __future__ import annotations`.

Reply:
Updated the type and went for the more generic `file` name.